include/sealib/compactarray.h

```diff
@@ -0,0 +1,48 @@
+#ifndef SEALIB_COMPACTARRAY_H_
+#define SEALIB_COMPACTARRAY_H_
+
+#include <math.h>
+#include <third-party/boost/include/boost/dynamic_bitset.hpp>
+
+typedef boost::dynamic_bitset<> Group;
+
+/**
+ * This compact array stores a given number of values in as low space as
+ * possible.
+ * Therefore an "epsilon" parameter is used: the smaller epsilon is, the larger
+ * the storage groups will be.
+ * @author Simon Heuser
+ */
+namespace Sealib {
+class CompactArray {
+ private:
+  std::out_of_range OUTOFBOUNDS = std::out_of_range("index out of bounds");
+
+ public:
+  /**
+   * Create a new compact array.
+   * @param count number of values this compact array can hold
+   * @param epsilon width parameter: 3/e should divide n
```

Comment:
Please rather floor the segment size such that it divides n.

Reply:
It is only a recommendation that 3/e should divide n. If not, the only side effect is that the last group will not be completely filled with values. I will add this info.